src/proxyp/Elements.cc

```diff
@@ -50,7 +50,8 @@ ProxyProtocol::IntegerToHeaderType(const SBuf &rawInteger)
     int64_t tlvType = 0;
 
     Parser::Tokenizer ptok(rawInteger);
-    if (!ptok.int64(tlvType, 10, false) || !ptok.atEnd())
+    const auto parsed = ptok.skip('0') || ptok.int64(tlvType, 10, false);
+    if (!parsed || !ptok.atEnd())
```

Comment:
This new code works (AFAICT), but it is very difficult to comprehend, and, from a leading zeros _writer_ pointer of view, it produces a rather strange exception -- such a writer is nearly guaranteed to think that `010` _is_ a decimal integer. Please refactor to better target leading zeros. Here is a sketch:

```
if (ptok.skip(0)) {
    if (!ptok.atEnd())
        throw TexcHere(ToSBuf("Invalid PROXY protocol TLV type value. ",
                              "Expected a decimal integer without leading zeros but got '",
                              rawInteger, "'")); // e.g., 077, 0xFF, or 0b101
    // tlvType stays zero
} else {
    Must(ptok.int64(tlvType, 10, false)); // the first character is a DIGIT
    if (!ptok.atEnd())
        throw TexcHere(ToSBuf("Invalid PROXY protocol TLV type value. ",
                              "Trailing garbage after ", tlvType, " in '",
                              rawInteger, "'")); // e.g., 1.0 or 5e0
}
```

Reply:
> Please refactor to better target leading zeros.

Done at d9f9458.